test/unit/host_test.rb

```diff
@@ -134,11 +134,20 @@ class HostTest < ActiveSupport::TestCase
   end
 
   test "sets compute attributes on create" do
-    Host.any_instance.expects(:set_compute_attributes).once.returns(true)
-    Host.create! :name => "myfullhost", :mac => "aabbecddeeff", :ip => "2.3.4.3",
+    h = Host.create! :name => "computeattributehost", :mac => "aabbecddeeff", :ip => "2.3.4.3",
       :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :medium => media(:one),
       :subnet => subnets(:two), :architecture => architectures(:x86_64), :puppet_proxy => smart_proxies(:puppetmaster),
-      :environment => environments(:production), :disk => "empty partition"
+      :environment => environments(:production), :disk => "empty partition", :compute_attributes => {:attr1 => 'blah'}
+    assert_equal 'blah', h.compute_attributes[:attr1]
+  end
+
+  test "should save compute attributes with indifferent access" do
+    h = Host.create! :name => "computeattributehost", :mac => "aabbecddeeff", :ip => "2.3.4.3",
+      :domain => domains(:mydomain), :operatingsystem => operatingsystems(:redhat), :medium => media(:one),
```

Comment:
Most of this probably isn't needed if you're only using .new and intending to test the initialiser.